File: ObjectAvoidance/java/src/main/java/elite/math/trig/triangle/Opposite.java
package elite.math.trig.triangle;

import elite.math.SquareRoot;

/**
 *
 * @author CFerg (Elite)
 */
public class Opposite {

    /**
     *
     */
    public double opposite;
    
    /**
     *
     */
    public Opposite(){
        this.opposite = 0.0;
    }
    
    /**
     *
     * @param a
     * @param h
     */
    public Opposite(Adjacent a, Hypotenuse h){
        this.opposite = new SquareRoot((h.hypotenuse * h.hypotenuse) - (a.adjacent * a.adjacent)).squareRoot;
    }
}

File: ObjectAvoidance/kotlin/src/elite/math/trig/triangle/Opposite.kt
package elite.math.trig.triangle

import elite.math.SquareRoot

/**
 *
 * @author CFerg (Elite)
 */
class Opposite {

    /**
     *
     */
    var opposite: Double = 0.toDouble()

    /**
     *
     */
    constructor() {
        this.opposite = 0.0
    }

    /**
     *
     * @param a
     * @param h
     */
    constructor(a: Adjacent, h: Hypotenuse) {
        this.opposite = SquareRoot(h.hypotenuse * h.hypotenuse - a.adjacent * a.adjacent).squareRoot
    }
}